### PR TITLE
Add Camera Mosaic plugin

### PR DESCRIPTION
**Pull request by jmeyers314**
### Description

Hello Stellarium developers!

As a long-time Stellarium fan and user as both an amateur and professional astronomer, I'm excited to submit a new plugin for your consideration. This plugin overlays the sensor outlines of mosaic cameras onto Stellarium's celestial sphere. While I do realize that both the oculars and telescope control plugins share some similarities, I hope you can appreciate the added detail that the mosaic provides.  I've found the plugin quite valuable in my recent work commissioning the brand new Vera C. Rubin Observatory.  Here are a couple quick screenshots of what the plugin provides:

Screenshot 2024-12-12 at 4 00 40 PM

Screenshot 2024-12-12 at 4 11 31 PM

Some key features are:

1. Includes several pre-configured cameras (LSSTCam, DECam, HSC, and MegaPrime)
2. Allows users to add custom cameras via JSON
3. Adjustable pointing and rotation either through plugin configuration or TCP messages

The plugin has garnered interest among my colleagues, already prompting a few of them to compile from my fork. A pre-compiled binary (i.e., distribution with the mainline Stellarium) would likely increase adoption significantly.

As this is my first Qt project, I welcome feedback on adherence to Qt and Stellarium conventions. I'm particularly interested in whether integration with the Stellarium remote control plugin would be an alternative to setting up a new TCP server for the plugin.

I'm grateful for Stellarium's impact on the astronomical community and hope this contribution aligns with your vision for the software.

Thanks,
Josh Meyers

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update
- [ ] Housekeeping

### How Has This Been Tested?

While there are no formal unit tests, I've used the plugin interactively for several weeks now, including as an observing aide while commissioning the Vera C. Rubin Observatory.

**Test Configuration**:
* Operating system: <MacOS, 14.7 Sonoma>
* Graphics Card: < Apple M3 Pro Integrated Graphics>

## Checklist:
- [ ] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Note: I will work on the above checklist imminently, but thought it would be a good idea to initiate this PR in the meantime.

### CODE REVIEW

File: plugins/MosaicCamera/src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     gui/MosaicCameraDialog.hpp
+     gui/MosaicCameraDialog.cpp
+)
+
+SET(MosaicCameraDialog_UIS
+     gui/mosaicCameraDialog.ui
+)
+
+SET(MosaicCamera_RES ../resources/MosaicCamera.qrc)
+QT5_ADD_RESOURCES(MosaicCamera_RES_RCC ${MosaicCamera_RES})
```
Comment (alex-w) [inline plugins/MosaicCamera/src/CMakeLists.txt:24]: The plugin cannot be compiled with Qt6 - please fix it (see other plugins for an example)

**Comment by alex-w** [comment]
Thank you very much for your plugin, but at the moment compilation is failed with Qt5 and Qt6 - please fix it.

Just after seeing the code: why TCP server is required in plugin and what will happen when TCP message will have wrong format?

File: plugins/MosaicCamera/src/gui/MosaicCameraDialog.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA  02110-1335, USA.
+*/
+
+// #include <QFileDialog>
+
+#include "StelApp.hpp"
+#include "ui_MosaicCameraDialog.h"
```
Comment (alex-w) [inline plugins/MosaicCamera/src/gui/MosaicCameraDialog.cpp:24]: Hint: `#include "ui_mosaicCameraDialog.h"`

**Comment by jmeyers314** [comment]
> Thank you very much for your plugin, but at the moment compilation is failed with Qt5 and Qt6 - please fix it.

Thanks, hopefully this is now addressed...

> Just after seeing the code: why TCP server is required in plugin and what will happen when TCP message will have wrong format?

I wanted there to be a way for an external program to set the camera mosaic ra/dec/rotation, and this was the first option I was able to get work.  I am very curious however if there's a way to integrate with the Remote Control plugin.  I couldn't figure it out at first glance.

Nothing will currently happen if the format is wrong.  You can experiment with, e.g.,

`echo "LSSTCam,-2.0,1.0,12.0" | nc localhost 5772`

which will set the LSSTCam overlay to ra=-2 deg, dec=1 deg, rotation = 12 deg.

On the other hand, 

`echo "LSSTCam,-2.0,1.0,12.0,blah" | nc localhost 5772`

will currently be silently ignored.

**Comment by alex-w** [comment]
The comma char is delimiter in numbers in some languages ;)

**Comment by alex-w** [comment]
Please check CodeFactor report

**Comment by alex-w** [comment]
> 2. Allows users to add custom cameras via JSON

But right now it's developers defined cameras - I guess this is a planning option. 

I think the version of plugin should be 0.0.0 :)

**Comment by gzotti** [comment]
Wow, that's quite a thing, Stellarium used for Vera Rubin etc... Not sure how many people would actually use the plugin, though. But it's also a great thing for outreach demos in the amateur communities.
Technically, it could be integrated as yet another instrument type for the Oculars plugin. However, that latter has some long overdue things to fix first, so let's stay with a separate plugin at least for now. 

I have not deeply looked at the code yet. You can indeed use the RemoteControl plugin to send mini scripts which just include one "setter" command or other "public slot" which are automatically exposed as scripting commands. Connecting to APT or my Unity bridge works like that. For this to work properly, settable variables must be exposed as PROPERTY. See other plugins for countless examples.

If your external application that communicates with Stellarium over RemoteControl needs to know Stellarium's state, there is a built-in mechanism to transmit, at first connection, the state of all StelProperty variables (which are basically the Qt PROPERTY entries), and later only transmit the small set of changes relevant to keep state in-sync. See scripting docs.

**Comment by jmeyers314** [comment]
> The comma char is delimiter in numbers in some languages ;)

Thanks, good point.  I'll switch this to a space for the moment.  Let me know if you have another suggestion though.

**Comment by jmeyers314** [comment]
> Wow, that's quite a thing, Stellarium used for Vera Rubin etc... Not sure how many people would actually use the plugin, though. But it's also a great thing for outreach demos in the amateur communities.

Yeah, potentially limited user interest did occur to me.  I played around with making a dynamic plugin for a while, but couldn't get it to work.  I think the outreach angle is real though.

> Technically, it could be integrated as yet another instrument type for the Oculars plugin.

I did look into using an Ocular for monitoring the pointing before diving in to making my own plugin.  It works okay, but I like having the individual sensor boundaries and also the ability to leave the boundaries fixed while I pan around the sky looking at what's nearby to the current pointing.

> You can indeed use the RemoteControl plugin to send mini scripts which just include one "setter" command or other "public slot" which are automatically exposed as scripting commands.

Thanks, I'll dig deeper here. Not opening an additional port certainly seems preferable if it can be avoided.

**Comment by alex-w** [comment]
Is it possible to add a short description for each cameras (file + GUI)?

**Comment by jmeyers314** [comment]
> Is it possible to add a short description for each cameras (file + GUI)?

Yes, will do.

**Comment by jmeyers314** [comment]
Also, looks like the hooks are already in place for updating the plugin without the TCP server.  Syntax is something like:

`curl -d 'code=MosaicCamera.updateMosaic("LSSTCam",-2.0,1.0,12.0)' http://localhost:8090/api/scripts/direct`

So I'll get to work on removing the TCP bits.

**Comment by gzotti** [comment]
Yes, this is the mini-script API. However, even cleaner is the /api/stelproperty service. It would circumvent the scripting module and allow synchronizing connected instances of Stellarium (RemoteSync plugin). You only have to make properties out of your plugin's instance variables. 

Hmm, I see you have your setters like "setRA(QString &cameraName, double RA)" and display several cameras at the same time. Then maybe the mini-script is the simplest to change to.  Another path would be to have a "currentCamera" property onto which property setters would apply. (This could be propagated via RemoteSync.) Like:

MosaicCamera.setCurrentCamera("LSSTCam");
MosaicCamera.setCurrentRA(ra):
MosaicCamera.setCurrentDec(dec);
MosaicCamera.setCurrentRotation(rot);

RA/dec could also be constructed and manipulated together as Vec3d type. See various examples.

**Comment by jmeyers314** [comment]
> However, even cleaner is the /api/stelproperty service.

Thanks, I managed to figure this out today.  Now the following works:

```
curl -d "id=MosaicCamera.currentCamera&value=LSSTCam" http://localhost:8090/api/stelproperty/set
curl -d "id=MosaicCamera.ra&value=-2.0" http://localhost:8090/api/stelproperty/set
curl -d "id=MosaicCamera.dec&value=1.0" http://localhost:8090/api/stelproperty/set
curl -d "id=MosaicCamera.rotation&value=12.0" http://localhost:8090/api/stelproperty/set
```

I kept the `setRA(const QString& cameraName, double ra)` style setters and getters for now alongside ones where you set `currentCamera` first, as I think they're more intuitive for maintaining multiple cameras.  But let me know what you think.

I think I'm happy with the interface now, but still have some cleanup/documentation work to go through.

**Comment by gzotti** [comment]
Yes, like that. Don't forget to document what angles ra/dec/rotation are: decimal degrees, decimal hours, radians, ...
No problem to keep the other setters like setRA(name, value) e.g. as higher-level scripting interface, as long as you then call the single property setters internally (to allow correct propagation).

**Comment by jmeyers314** [comment]
While fixing up the cameras I've had another related plugin idea: outlining astronomical survey boundaries.  For example, outlining the Hubble Ultra Deep Field, or the Dark Energy Survey footprint.  User fields, e.g., "all the locations of my astrophotos" could be implemented too.  In fact, user input here seems essential, as there are too many cameras and surveys to include all by default.

One challenge to consider is that larger regions might require more than a simple outline; some form of shading could be necessary to clearly distinguish between interior and exterior areas.

Bringing this up now because if this feature were to be included, renaming the plugin to something like "Footprints" or "Overlays" might be appropriate, as it would encompass both camera and survey boundaries. Thoughts appreciated.

**Comment by gzotti** [comment]
Should these footprints not just be HiPS surveys? In any case, I don't see them in this plugin. In essence, it's like "footprint shapefiles" in GIS, we currently don't have such polygonal functionality.

**Comment by gzotti** [comment]
Looks good for experts. However, if we want to have that in the package, 
- The plugin should have an activation button in the lower button bar. A left-click enables showing the active grids, a right-click calls up the plugin dialog.
- we need a chapter in the User Guide. Probably at the end of plg_atTheTelescope.tex, we can have a new subsection. I see you have added a README, this can be base for this chapter. In here we expect to find some motivation text (0-2 lines why I'd ever want to use the plugin), usage instructions, configuration instructions (elements of your JSON) and a list of default-provided instruments. Here I would expect info about the color-coding of the LSSTCam fields.  Hyperlinks of external websites are OK for "further reading", and any scientific references can be added to the biblatex file guide.bib. 
- I suggest a minimum online documentation in the "About" tab as well: plugin purpose, and a list of the default-configured instruments. You can provide e.g. HTML hyperlinks to info pages or websites of the actual instruments. 
- When one wants to add own instruments, the "private" path of a plugin where it may find data is `$USER_DIR/modules/<pluginName>`. This avoids the user editing system-protected files in the installation directory. If you want to make the instrument list user-expandable, it is OK to test if a JSON resides there and use it instead of the default file in the installation directory, or else to even mix default and user-configured JSON files (possibly overwriting entries by user-defined ones of the same name).

**Comment by jmeyers314** [comment]
Hey, sorry for the silence.  It's been a chaotic start to the year.

But to reply to some of the more recent points:

> Should these footprints not just be HiPS surveys? In any case, I don't see them in this plugin. In essence, it's like "footprint shapefiles" in GIS, we currently don't have such polygonal functionality.

Ah, I hadn't actually dived into the HiPS section of Stellarium before.  Yeah, that covers at least most of my use case.

> The plugin should have an activation button in the lower button bar. A left-click enables showing the active grids, a right-click calls up the plugin dialog.

Sound good.  Guess I'll be trying out icon design for the first time here too :)

> we need a chapter in the User Guide. 

Yep, everything here sounds good.  Will do.

> I suggest a minimum online documentation in the "About" tab as well

:+1:

> When one wants to add own instruments, the "private" path of a plugin where it may find data is $USER_DIR/modules/<pluginName>. This avoids the user editing system-protected files in the installation directory. If you want to make the instrument list user-expandable, it is OK to test if a JSON resides there and use it instead of the default file in the installation directory, or else to even mix default and user-configured JSON files (possibly overwriting entries by user-defined ones of the same name).

I may have more questions on this one when I get into implementing it, but sounds good in principle.

Thanks for your patience.  I'm still quite interested in getting this done.

**Comment by alex-w** [comment]
@jmeyers314 is the plugin ready to review?

**Comment by jmeyers314** [comment]
Hi again,
  I think this is ready for review now (assuming it passes the CI checks).  Looking forward to hearing back comments :)
-Josh

File: guide/plg_atTheTelescope.tex [hunk 3]
```diff
@@ -1153,7 +1156,189 @@ \subsection*{Author}
...
+\subsection{Introduction}
+\label{sec:plugins:MosaicCamera:intro}
+This plugin overlays an outline of one or more mosaic cameras, i.e., cameras with multiple sensors, onto the Stellarium celestial sphere.
+With this functionality, one may compare the fields of view of different cameras, including potentially one's own astrophotography setup.
+Combined with the scripting and remote control features of Stellarium, various visualizations are possible, including real-time tracking of the observations of telescopes, playback of past observations, or public outreach demonstrations of the capabilities of various instruments.
+
+\subsection{Usage}
+\label{sec:plugins:MosaicCamera:usage}
+
+To use the plugin, first enable it by selecting the "Load at startup" checkbox under the "Plugins" tab of the main Configuration window.
```
Comment (gzotti) [inline guide/plg_atTheTelescope.tex:1179]: Please use "grave accents" to start a quotation, and apostrophes to end. Like  ` ``Plugins'' `
Also, IMHO the first sentence is not needed. General plugin handling is known or described elsewhere.

File: guide/plg_atTheTelescope.tex [hunk 4]
```diff
@@ -1153,7 +1156,189 @@ \subsection*{Author}
...
+
+\subsection{Cameras}
+\label{sec:plugins:MosaicCamera:cameras}
+Multiple cameras can be individually configured and displayed.
+
+\subsubsection{Default Cameras}
+\label{sec:plugins:MosaicCamera:cameras:default}
+A few cameras come pre-packaged with Stellarium.  These are:
+
+\subsubsubsection{\textbf{LSSTCam}}
```
Comment (gzotti) [inline guide/plg_atTheTelescope.tex:1191]: This depth is not available. You can use 

`\paragraph{LSSTCam}`

and please without additional textbf.

File: guide/plg_atTheTelescope.tex [hunk 5]
```diff
@@ -1153,7 +1156,189 @@ \subsection*{Author}
...
+Camera sensors are specified via their corners, which are assumed to form spherical polygons.
+The corners should be specified as \href{https://en.wikipedia.org/wiki/Gnomonic_projection}{gnomonic projections} from the sphere to a tangent plane in radians.
+Multiple groups of sensors can compose the same camera.
+For example, LSSTCam has separate sensor groups for wavefront sensors, guider sensors, and each of two types of science sensors.
+
+\newpage
+
+Each camera is defined via a JSON file with the structure:
+
+\begin{script}
```
Comment (gzotti) [inline guide/plg_atTheTelescope.tex:1241]: We have definitions for json:

```
\begin{jsonfile}[\scriptsize]
...
\end{jsonfile}
```

File: plugins/MosaicCamera/resources/Latiss.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      "comment": "Yellow",
+      "value": [
+        1.0,
+        1.0,
+        0.0,
+        1.0
+      ]
+    }
+  }
+]
```
Comment (gzotti) [inline plugins/MosaicCamera/resources/Latiss.json:34]: Please add a newline

File: plugins/MosaicCamera/src/MosaicCamera.cpp [hunk 7]
```diff
@@ -0,0 +1,560 @@
...
+MosaicCamera::~MosaicCamera()
+{
+    delete configDialog;
+}
+
+void MosaicCamera::loadSettings()
+{
+    conf->beginGroup("MosaicCamera");
+    currentCamera = conf->value("currentCamera").toString();
+    setFlagShowButton(conf->value("showButton").toBool());
```
Comment (gzotti) [inline plugins/MosaicCamera/src/MosaicCamera.cpp:89]: You can safely add a default-true here:

`setFlagShowButton(conf->value("showButton", true).toBool());`

File: plugins/MosaicCamera/src/MosaicCamera.cpp [hunk 8]
```diff
@@ -0,0 +1,560 @@
...
+        flagShowMosaicCamera = b;
+        emit flagMosaicCameraVisibilityChanged(b);
+    }
+}
+
+void MosaicCamera::initializeUserData()
+{
+    userDirectory = StelFileMgr::getUserDir()+"/modules/MosaicCamera/";
+    QDir dir(userDirectory);
+    if (!dir.exists())
```
Comment (gzotti) [inline plugins/MosaicCamera/src/MosaicCamera.cpp:177]: Maybe also test for empty dir, when user only deletes contents in an attempt to reset.

File: plugins/MosaicCamera/src/MosaicCamera.cpp [hunk 9]
```diff
@@ -0,0 +1,560 @@
...
+    StelPainter painter(prj);
+
+    Vec3d startPoint, endPoint;
+
+    for (const auto& camera : cameras)
+    {
+        if (!camera.visible)
+            continue;
+
+        double alpha = camera.ra / 57.29577951308232 + 1.5707963267948966;
```
Comment (gzotti) [inline plugins/MosaicCamera/src/MosaicCamera.cpp:486]: Please use the usual named constants from StelUtils.hpp. M_PI_2 ~ 1.57..., M_180_PI ~ 57.29...

File: plugins/MosaicCamera/src/MosaicCamera.hpp [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+	 * setting and getting properties through the Stellarium Remove Control HTTP API.
+     */
+
+	/// @property enabled
+	/// @brief Are mosaic camera overlays enabled?
+	Q_PROPERTY(bool enabled		     READ isEnabled              WRITE enableMosaicCamera  NOTIFY flagMosaicCameraVisibilityChanged)
+
+	Q_PROPERTY(bool showButton       READ getFlagShowButton      WRITE setFlagShowButton   NOTIFY flagShowButtonChanged)
+
+	/// @property currentCamera
```
Comment (gzotti) [inline plugins/MosaicCamera/src/MosaicCamera.hpp:75]: The StelProperties fully work only when NOTIFY signals are used. Please add them like 

`... NOTIFY currentRAChanged)`

File: plugins/MosaicCamera/src/gui/MosaicCameraDialog.cpp [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+	html += "<h4>" + q_("DECam") + "</h4>";
+	html += "<p>" + q_("The Dark Energy Camera. A 570 megapixel camera mounted on the Blanco telescope at the Cerro Tololo Inter-American Observatory.  See more at: ") + "<a href=\"https://www.darkenergysurvey.org/the-des-project/instrument/\">https://www.darkenergysurvey.org/the-des-project/instrument/</a>" + "</p>";
+	html += "<h4>" + q_("MegaPrime") + "</h4>";
+	html += "<p>" + q_("A 378 megapixel camera mounted on the Canada-France-Hawaii Telescope at Mauna Kea Observatory.  See more at: ") + "<a href=\"https://www.cfht.hawaii.edu/Instruments/Imaging/MegaPrime/\">https://www.cfht.hawaii.edu/Instruments/Imaging/MegaPrime/</a>" + "</p>";
+	html += "<h4>" + q_("Latiss") + "</h4>";
+	html += "<p>" + q_("The LSST Atmospheric Transmission Imager and Slitless Spectrograph.  A 16 megapixel camera mounted on the Rubin Observatroy Auxiliary telescope.  See more at: ") + "<a href=\"https://noirlab.edu/public/programs/vera-c-rubin-observatory/rubin-auxtel/\">https://noirlab.edu/public/programs/vera-c-rubin-observatory/rubin-auxtel/</a>" + "</p>";
```
Comment (gzotti) [inline plugins/MosaicCamera/src/gui/MosaicCameraDialog.cpp:197]: Typo: "Rubin Observatory"

**Comment by gzotti** [comment]
I just wondered where the grids are. All reset to zero.  All behind the Sun?  Oh, sure, March 20... :-)

I think to become visible, the button should be visible by default. And if you can, I'd like to see a button "center on selected object". 

In all, this is really solid and great for outreach and explanations what the latest big telescopes are used for.

**Comment by jmeyers314** [comment]
Thanks for the review.  I think I addressed all of the the in-line comments above.

> I just wondered where the grids are. All reset to zero. All behind the Sun? Oh, sure, March 20... :-)

:)

> I think to become visible, the button should be visible by default. And if you can, I'd like to see a button "center on selected object".

The visible by default bit is addressed in your `setFlagShowButton(conf->value("showButton", true).toBool());` comment above I think?  Or did you mean something different?

Adding a "center on selected object" button sounds good.  Do you mean to place this within the configuration dialog, or somewhere else?

> In all, this is really solid and great for outreach and explanations what the latest big telescopes are used for.

Thanks!

**Comment by gzotti** [comment]
> Thanks for the review. I think I addressed all of the the in-line comments above.
> 
> > I think to become visible, the button should be visible by default. And if you can, I'd like to see a button "center on selected object".
> 
> The visible by default bit is addressed in your `setFlagShowButton(conf->value("showButton", true).toBool());` comment above I think? Or did you mean something different?

Yes, that should be all that's required.


> Adding a "center on selected object" button sounds good. Do you mean to place this within the configuration dialog, or somewhere else?

Yes, right in this dialog. Just add a button that can trigger "put selected grid onto the currently selected object." If there is no selection, it shall do nothing.

**Comment by jmeyers314** [comment]
>Yes, right in this dialog. Just add a button that can trigger "put selected grid onto the currently selected object." If there is no selection, it shall do nothing.

Cool, I think I got this working now too.

**Comment by gzotti** [comment]
Works great! 

Now, playing for a few minutes, the reverse application question comes up. "Where is my Latiss?" (It's actually quite small, so easily overlooked. And BTW, is it really not capitalized?) 
If you also have this question, another button "center on current camera" (`StelMovementMgr::moveToJ2000(const Vec3d& aim, const Vec3d& aimUp, float moveDuration, ZoomingMode zooming)`maybe adjusting field of view with something like `StelMovementMgr::setFOVDeg(5 * FOV(currentCamera))`?) would be the solution. 


I don't know, it would of course be extra eyecandy to attach such a mosaic layout to the telescope control plugin's telescope aim icon, but that's probably a long-time aim.

**Comment by jmeyers314** [comment]
>Now, playing for a few minutes, the reverse application question comes up. "Where is my Latiss?" (It's actually quite small, so easily overlooked. And BTW, is it really not capitalized?)

Nice catch.  It should be capitalized here.  (I'm used to staring at it as `Latiss` for uninteresting coding standards reasons....)

>If you also have this question, another button "center on current camera" (StelMovementMgr::moveToJ2000(const Vec3d& aim, const Vec3d& aimUp, float moveDuration, ZoomingMode zooming)maybe adjusting field of view with something like StelMovementMgr::setFOVDeg(5 * FOV(currentCamera))?) would be the solution.

Added this.  The `moveToJ2000` bit works nicely (smooth animation).  The `setFov` bit though is instantaneous.  Is there a zoom movement that's smoother I could use?  If not, I'd be tempted to only do the J2000 bit and leave the zoom in/out bit to the user.

> I don't know, it would of course be extra eyecandy to attach such a mosaic layout to the telescope control plugin's telescope aim icon, but that's probably a long-time aim.

Sure, that makes a lot of sense, but agree it's probably a future PR.

**Comment by gzotti** [comment]
> The `setFov` bit though is instantaneous. Is there a zoom movement that's smoother I could use? If not, I'd be tempted to only do the J2000 bit and leave the zoom in/out bit to the user.

Ah sorry, sure. `StelMovementMgr::zoomTo(fov, time_sec)`

**Comment by jmeyers314** [comment]
> Ah sorry, sure. StelMovementMgr::zoomTo(fov, time_sec)

Nice.  That's much better.

File: plugins/MosaicCamera/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,14 @@
+# This is the cmake config file for the MosaicCamera
+SET(MOSAICCAMERA_VERSION "0.0.0")
```
Comment (alex-w) [inline plugins/MosaicCamera/CMakeLists.txt:2]: Why version is 0.0.0?

File: guide/plg_atTheTelescope.tex [hunk 13]
```diff
@@ -1153,7 +1156,187 @@ \subsection*{Author}
...
+
+\subsection{Control}
+\label{sec:plugins:MosaicCamera:control}
+
+There are several ways to control the Mosaic Camera plugin.
+
+\subsubsection{Mosaic Camera Settings Dialog}
+\label{sec:plugins:MosaicCamera:control:dialog}
+
+The Mosaic Camera settings dialog can be opened by right-clicking the MosaicCamera icon, or by selecting "configure" from the Plugins tab of the main Stellarium Configuration window.
```
Comment (alex-w) [inline guide/plg_atTheTelescope.tex:1278]: It should be _``configure''_ here

File: plugins/MosaicCamera/src/MosaicCamera.hpp [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+	void initializeUserData();
+	void copyResourcesToUserDirectory();
+	void setCameraFieldDiameter(Camera& camera);
+	static double gnomonicChordSeparationSquared(const QPointF& p1, const QPointF& p2);
+
+	StelGui* gui;
+	MosaicCameraDialog* configDialog;
+	StelButton* toolbarButton;
+	StelCore* core;
+
```
Comment (alex-w) [inline plugins/MosaicCamera/src/MosaicCamera.hpp:181]: Please remove blank line here

File: plugins/MosaicCamera/CMakeLists.txt (hunk 12 above)
Comment (gzotti) [reply plugins/MosaicCamera/CMakeLists.txt:2]: The simplest solution is 

`SET(MOSAICCAMERA_VERSION "${VERSION}")
`
It just grows with the program. (The separate version numbers comes from long time ago, when "dynamical plugins" were really external downloads.)

File: guide/plg_atTheTelescope.tex (hunk 13 above)
Comment (jmeyers314) [reply guide/plg_atTheTelescope.tex]: Good catch.  Thanks.

**Review by alex-w — APPROVED** [review]
Good work, thanks!

**Review by gzotti — APPROVED** [review]
Very nice indeed! Also good docs, thank you! While only few users will have their own mosaic cameras, this is at least an interesting eyecatcher for indoor discussion. I am still flabbergasted about the origins of this plugin! Any related open-access papers could be later added (linked) to the online help or User Guide pages.

**Comment by jmeyers314** [comment]
Thanks both for all the help getting this merged!  I'm very happy to be able to contribute to this open source project I've enjoyed using for many many years.  I'd be happy to add links to any open-access papers as they become available. And if there are ever any bugs or questions regarding the plugin, please feel free to ping me.

One final question: Can you share some details about the Stellarium release cycle?  The Rubin Observatory is expected to resume sky observations and begin commissioning the full 3.2-gigapixel camera in approximately three weeks. If there's a possibility of releasing a new version with the plugin within that timeframe, I'd be very appreciative (and so would many of our observing and commissioning staff). Though, of course, I understand if this is too big an ask to consider.  Thanks again!

**Comment by 10110111** [comment]
Stellarium is released 4 times a year, around equinoxes and solstices. But I think there'll be some [development snapshots](https://github.com/Stellarium/stellarium-data/releases/tag/weekly-snapshot) available (called weekly but published a bit less often).

**Comment by jmeyers314** [comment]
> Stellarium is released 4 times a year, around equinoxes and solstices. But I think there'll be some [development snapshots](https://github.com/Stellarium/stellarium-data/releases/tag/weekly-snapshot) available (called weekly but published a bit less often).

Actually, I see development notes and names of installers here, but can't find any actual links/assets.  Am I missing something?

**Comment by gzotti** [comment]
With 25.1 out just on last Sunday, there are no newer weekly builds.

**Comment by 10110111** [comment]
> Actually, I see development notes and names of installers here, but can't find any actual links/assets.

Hmm, interesting. I'd guess that they've been removed on publishing of the latest stable release. Should appear again the next time a snapshot is published. @alex-w Am I right?

**Comment by alex-w** [comment]
> > Actually, I see development notes and names of installers here, but can't find any actual links/assets.
> 
> Hmm, interesting. I'd guess that they were removed on publishing of the latest stable release. Should appear again the next time a snapshot is published. @alex-w Am I right?

Yes, changes in the list of changes and new packages will coming in upcoming weekend. Plus notifications will be added in issues and pull requests, which was closed at the moment of publishing for related snapshot and has target to the next release  :)

**Comment by jmeyers314** [comment]
So the details definitely did not come through, but thought you all might appreciate [this photo](https://static01.nyt.com/images/2025/06/20/multimedia/00HS-sci-rubin-data-05-fgwm/00HS-sci-rubin-data-05-fgwm-superJumbo.jpg?quality=75&auto=webp) of the Rubin Observatory control room published in the NY Times today.  You can see Stellarium running on one of the giant TV monitors.

From [this article](https://www.nytimes.com/2025/06/20/science/vera-rubin-telescope-data.html?unlocked_article_code=1.QU8.YyU1.vhUSgWEqEYX7&smid=url-share).
